internal/provider/resource_gitlab_group_project_file_template_test.go

```diff
@@ -0,0 +1,139 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+	"testing"
+)
+
+func TestAccGitlabGroupProjectFileTemplate_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	var parentGroup gitlab.Group
+	var templateProject gitlab.Project
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckProjectFileTemplateDestroy,
+		Steps: []resource.TestStep{
+			{
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(rInt, "firstProject"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupExists("gitlab_group.parent_group", &parentGroup),
+					testAccCheckGitlabProjectExists("gitlab_project.template_project", &templateProject),
+					// Note - we can't use the testAccCheckGitlabGroupAttributes, because that checks the TF
+					// state attributes, and file project template explicitly doesn't exist there.
+					testAccCheckGitlabGroupFileTemplateValue(&parentGroup, &templateProject),
+				),
+			},
+			{
+				//Test that when we update the project name, it re-links the group to the new project
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateConfig(rInt, "secondProject"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupExists("gitlab_group.parent_group", &parentGroup),
+					testAccCheckGitlabProjectExists("gitlab_project.template_project", &templateProject),
+					testAccCheckGitlabGroupFileTemplateValue(&parentGroup, &templateProject),
+				),
+			},
+			{
+				//Test that when we remove the resource, the group gets it's value set to 0
+				SkipFunc: isRunningInCE,
+				Config:   testAccGroupProjectFileTemplateRemovedConfig(rInt),
```

Comment:
When the test project and group is outside of TF as per my other comment we can just rely on the normal post-destroy checks ...